### DISCUSSION TITLE
running custom commands upon lock and unlock

### DISCUSSION THREAD

**Discussion by lestephane** [rcaelers/workrave#453] [Q&A]
I use custom cli commands to punch in and out of tasks.
When workrave locks everything, I no longer can punch out with the keyboard...
If I punch out in time before workrave locks everything,  I then forget to punch in after workrave unlocks.
It would seem to me that letting the user run custom commands upon lock and unlock is the best approach.
Is there a way to achieve this?

There might be larger narrative about how workrave integrates with external punch clocks here (at least how it automatically punches out the user), though it is not my intent to start such as design discusssion. I want to squarely focus on allowing the user to have workrave execute arbitrary commands upon workrave lifecycle events.

**Comment by igorsantos07** [comment]
This would be useful as well for other stuff, like stopping music to play. I got quite addicted to that feature when I used `tomate-gtk`, actually. My "brain background task" tends to notice the lack of music after a while, and that worked very well as a gentle reminder it's time to leave the computer for a bit - instead of intrusive reminders popping up on my face while I'm focused.
